### ISSUE TITLE
ConditionDamage - Damage calculations are incorrect

### ISSUE THREAD

**Issue by Quintinon** [otland/forgottenserver#655]
### TLDR

There are a few issues in ConditionDamage in conditions.cpp.
1. ConditionDamage::getTotalDamage() is incorrect when damageList is empty.
2. ConditionDamage instances that use minDamage and maxDamage with startDamage != 0 do not always do damage between minDamage and maxDamage.
### Background

I was reviewing #618 and noticed something I didn't quite understand in conditions.cpp, in the ConditionDamage class.
### Issue Number 1

In ConditionDamage::getTotalDamage() when damageList is empty I was expecting the average of minDamage and maxDamage, but instead found this:

``` cpp
result = maxDamage + (maxDamage - minDamage) / 2;
```

Which is maxDamage plus average of maxDamage and minDamage.
##### Background on this issue

I was inspecting ConditionDamage::generateDamageList() when i stumbled upon this section of the code:

``` cpp
x1 = std::fabs(1.0 - ((static_cast<float>(sum)) + i) / med);
x2 = std::fabs(1.0 - (static_cast<float>(sum) / med));
```

Because I did not immediately understand how to analyze this abstractly, I wrote a c# program to do numerical analysis on the ConditionDamage::int() which calls ConditionDamage:generateDamageList().
##### Issue itself

I found that if startDamage = 0, minDamage = 10, maxDamage = 100, then the average damage was 55 like I expected. However, if startDamage > 4, then the average starts to float upwards (that is, in a sample size of 100k  iterations, the damage range went from [10,100] to [15,100]).
There is a 'safeguard' in place where:

``` cpp
if (startDamage > maxDamage) {
    startDamage = maxDamage;
} else if (startDamage == 0) {
    startDamage = std::max<int32_t>(1, std::ceil(amount / 20.0));
}
```

but if I set StartDamage = 100, then every time I get 5050 as the damage done (total).

Things get worse the farther apart minDamage and maxDamage are, and if minDamage = 0, then things can get weird too.
### Conclusion

I suggest that someone with more time than I have review the formulas in ConditionDamage::generateDamageList() and ConditionDamage::init() and correct them.
From this, ConditionDamage::getTotalDamage() can be re-written to be correct.

I should be able to provide mathematical assistance if needed, but I do not have the source code setup where I can compile or make changes to the forgotten server.

Source code for my test app can be found here:
https://dl.dropboxusercontent.com/u/186942/DamageConditionTest.zip
##

**Comment by WibbenZ** [comment]
Seeing as no one has helped to confirm this issue im closing it.
Feel free to inform us if the issue is still present.
